code/modules/wiremod/port.dm

```diff
@@ -52,185 +53,165 @@
 			return "grey"
 
 /datum/port/Destroy(force)
+	disconnect_all()
 	connected_component = null
 	return ..()
 
 /**
- * Returns the value to be set for the port
- *
- * Used for implicit conversions between outputs and inputs (e.g. number -> string)
- * and applying/removing signals on inputs
+ * Sets the port's value to v.
+ * Casts to the port's datatype (e.g. number -> string), and assumes this can be done.
  */
-/datum/port/proc/convert_value(prev_value, value_to_convert)
-	if(prev_value == value_to_convert)
-		return prev_value
-	. = value_to_convert
-
-	if(isnull(value_to_convert))
-		return null
+/datum/port/proc/put(v)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Name this new_value, or name it value and use `src.value` when using/updating the existing variable.

Reply:
I also don't think `put` is a better name at all and am not sure why you renamed `set_output`.

Reply:
Yeah I think put is a very confusing name.

Reply:
`set_value` was taken, `put` seemed better than `port_set_value`.

Reply:
False positive, `set_value` wasn't taken! May I use that instead of `set_output` and `set_input`? Or would you rather like there to be `set_output` and `set_input` aliases that typecheck for the port actually being of that kind?

Reply:
Not too sure about aliases, though I guess `set_value` should be fine? Just makes diffs and other PRs on this annoying.